### ISSUE TITLE
Icons from icon files saved with ImageGlass appear blank

### ISSUE THREAD

**Issue by SilverKnight-Z** [d2phap/ImageGlass#1263]
### System information:
- Windows OS version: Windows 7 Starter SP1 32-bit 

- ImageGlass version: 8.3.11.21 (x86 Installer) 

### To Reproduce
- Open ImageGlass.
- Open an image.
- Save the image as ICO file.
- Open the Windows explorer.
- Check the folder where you saved the ICO file.
- Change the folder view to details view, list view or small icons view.
- Check if the preview of the saved icon appears in any of the above folder views.

### Actual behavior:
The saved icon file appears blank in detail, list and small icon views in Windows file explorer.

### Expected behavior:
The saved icon preview appears correctly in all Windows Explorer folder views.

### Screenshots:
Content view
![20220123_145808.jpg](https://user-images.githubusercontent.com/82909608/150693543-62cfc29b-cc69-4ab5-952b-c89a4ea2b24c.jpg)

Tile view
![20220123_145859.jpg](https://user-images.githubusercontent.com/82909608/150693631-40a8ec62-0763-4f0a-80f3-68823ac2422a.jpg)

List view
![20220123_145929.jpg](https://user-images.githubusercontent.com/82909608/150693649-99baaa04-c8cb-4969-8fd7-c488ce5aabad.jpg)

Small icons view
![20220123_145954.jpg](https://user-images.githubusercontent.com/82909608/150693787-2da8d253-d515-487e-856a-ab86e749262e.jpg)

Mid icons view
![20220123_150017.jpg](https://user-images.githubusercontent.com/82909608/150693771-effc68a7-09e9-41aa-afac-c10847cc7de4.jpg)

Large icons view
![20220123_150111.jpg](https://user-images.githubusercontent.com/82909608/150693671-795cc065-f7c5-4c0a-bc0f-763a4e59fb2d.jpg)

Very large icons view
![20220123_145954.jpg](https://user-images.githubusercontent.com/82909608/150693660-030cbda8-60b3-4066-8efa-d17d16e88107.jpg)

### Additional context:
The preview of the saved icon works correctly in the other folder views and the Windows Explorer status bar.

**Comment by fire-eggs** [comment]
Interesting. I can't repro with Win7 Pro 64 bit / IG 8.5.1.22.

I'll see if I can dig up a 32-bit Win-7 installer ...

**Comment by SilverKnight-Z** [comment]
I forgot to mention: It also happens with IG 8.5.1.22. I updated yesterday to see if this bug was fixed, but sadly it wasn"t. I also did I clean install but it's the same

**Comment by fire-eggs** [comment]
TL;DR : appears to be expected behavior from a clean Windows 7 Explorer. It would be nice if IG _could_ make a better ICO file. Also a reminder: IG doesn't provide the icons shown in Windows Explorer, a shell extension such as [SageThumbs](https://www.cherubicsoft.com/en/projects/sagethumbs/) might solve the issue.


Here's what I've found. This is with a clean Windows 7-SP1 x86 install.

Any .ico file saved by IG _cannot_ be seen in Windows Explorer (icon view), _nor_ in the Explorer Preview panel, _nor_ in the Explorer status bar (any Windows Explorer view).

[Note that a "real" .ico file (e.g. as provided in the IG install in the "Ext-Icons" folder) _cannot_ be seen in Preview panel etc, _either_. But they do show in the Windows Explorer icon view.]

At this point I have to guess you have an app, codec or other extension installed that lets you see these files in the Preview panel. I probably do too in my Windows 7-SP1 64bit setup.

The .ico files which are saved by IG don't provide all the different ICO sizes. That is, an .ICO file typically contains multiple icons: 16x16, 32x32, etc and possibly a final "extra extra large" PNG. An ICO file saved by IG doesn't include the smaller ICOs, _only_ the final PNG. It appears Windows Explorer in a "clean" setup won't let you view PNG-only .ICO files.

It would be nice if IG could save a composite ICO file containing at least one of the smaller ICO files. I'm not sure if that makes sense for a large original image but why not: a 96x96 thumbnail is perfectly viewable. I can't remember if this is a Magick.NET limitation or one in IG proper.

**Comment by SilverKnight-Z** [comment]
> At this point I have to guess you have an app, codec or other extension installed that lets you see these files in the Preview panel. I probably do too in my Windows 7-SP1 64bit setup.

In fact my Windows file explorer does not have any extensions, codecs or anything like that. What I did install were all the updates available for Windows 7 back in June 2020. Which reminded me that before I installed them, the thumbnails in some of my video files weren't appearing (video files with codecs and containers that Windows can play perfectly well, of course), and after installing the updates, that was completely resolved. 

Maybe that's why I can see icons where your 32-bit Windows can't, and I'm guessing your 64-bit Windows is fully updated and that's why you can see icons in that version. Maybe your 32-bit Windows is missing an update, unfortunately I don't know which one.

> The .ico files which are saved by IG don't provide all the different ICO sizes. That is, an .ICO file typically contains multiple icons: 16x16, 32x32, etc and possibly a final "extra extra large" PNG. An ICO file saved by IG doesn't include the smaller ICOs, only the final PNG. It appears Windows Explorer in a "clean" setup won't let you view PNG-only .ICO files.

That would explain what is happening in #1264; because inside the ".ico" container there is no icon, only an image.

> It would be nice if IG could save a composite ICO file containing at least one of the smaller ICO files. I'm not sure if that makes sense for a large original image but why not: a 96x96 thumbnail is perfectly viewable. I can't remember if this is a Magick.NET limitation or one in IG proper.

That would be really great, although I wasn't trying to save an icon with a large base image, but a 130x140 image.

So...
This issue shouldn't be about a bug, but a feature request?

**Comment by fire-eggs** [comment]
I'll mark it as "improvement" ...

**Comment by SilverKnight-Z** [comment]
Ok, thank you!
Edit: I'll close #1264 now that I now what's happening there

**Comment by d2phap** [comment]
I will close this because ImageGlass 9 does not support Windows 7 anymore
